packages/block-library/src/cover/style.scss

```diff
@@ -30,19 +30,33 @@
 
 	&.has-background-dim::before {
 		content: "";
+		background-color: inherit;
+	}
+
+	&.has-background-dim:not(.has-background-gradient)::before,
+	.wp-block-cover__gradient-background {
 		position: absolute;
 		top: 0;
 		left: 0;
 		bottom: 0;
 		right: 0;
-		background-color: inherit;
-		opacity: 0.5;
 		z-index: z-index(".wp-block-cover.has-background-dim::before");
 	}
 
+	&.has-background-dim:not(.has-background-gradient)::before,
+	& .wp-block-cover__gradient-background {
+		opacity: 0.5;
+	}
```

Comment:
Why this style exists in the first place (the opacity)? Do we want all background colors to have it?